lms/static/scripts/frontend_apps/config.js

```diff
@@ -1,7 +1,121 @@
 import { createContext } from 'preact';
 
+/**
+ * Parameters for an "canned" API call that the frontend can make to the server.
+ *
+ * The parameters of the call are decided by the backend, but the frontend
+ * decides _when_ to make the call and what to show while waiting for the
+ * response.
```

Comment:
For the various objects here I have included a short overview of what the parameter is and the properties of each. However I haven't gone into detail about what all the properties are as that information is contained in the `JSConfig` class in the backend, as mentioned in the note for the `ConfigObject` type below.